### ISSUE TITLE
Add libmpv

### ISSUE THREAD

**Issue by g-fb** [flathub/shared-modules#284]
It takes a while to build and there are ~20 apps using it https://github.com/search?q=org%3Aflathub%20libmpv&type=code

**Comment by hfiguiere** [comment]
> It takes a while to build

That's not an argument.

**Comment by Erick555** [comment]
It has tons of deps and without auto-updating it's rather unmaintainable as shared-module

**Comment by TingPing** [comment]
It is quite complex and regularly updating, which makes it both a good and bad idea.

I try to keep the Celluloid one updated and it looks like none of the others are up to date. So sharing that work would be nice.

**Comment by Erick555** [comment]
for Celluloid bot do updates for shared-modules it won't. Also bumping shared-modules in app repo is rare thing.

**Comment by g-fb** [comment]
If the lack of auto updating dependecies is not possible for shared modules, then how about having a regular flatpak package that is auto updated and then synced here?

> looks like none of the others are up to date

There are 8 other apps with up to date mpv

[haruna](https://github.com/flathub/org.kde.haruna/blob/15ba902ae16fc60fa59e61dd99b7d51ad52c02e7/org.kde.haruna.yml#L53
)
[Celluloid](https://github.com/flathub/io.github.celluloid_player.Celluloid/blob/c1dc8b7d6267104b37bbfb48904ce46d5150da34/io.github.celluloid_player.Celluloid.yaml#L62)
[rssguard](https://github.com/flathub/io.github.martinrotter.rssguard/blob/3db73a294194a5cb00cbee33a7ec235d7fad0c03/io.github.martinrotter.rssguard.yml#L39)
[PhotoQt](https://github.com/flathub/org.photoqt.PhotoQt/blob/1bea2dd9dffcec15fbf1e0089f6bed008a2f1749/org.photoqt.PhotoQt.yml#L283)
[jellyfin-media-player](https://github.com/flathub/com.github.iwalton3.jellyfin-media-player/blob/70a222b48d6cb91141b0131aee9da26a8d9fc38d/com.github.iwalton3.jellyfin-media-player.json#L84)
[mugen](https://github.com/flathub/moe.karaokes.mugen/blob/651ff6f095f6124a22e783adc8c6dca5e4934160/moe.karaokes.mugen.yml#L442)
[KikoPlay](https://github.com/flathub/io.github.KikoPlayProject.KikoPlay/blob/9119f7322b33b26f2a54311d2b314de8582029b8/io.github.KikoPlayProject.KikoPlay.yaml#L64)
[hydrus](https://github.com/flathub/io.github.hydrusnetwork.hydrus/blob/172e235e6b877157b9755a67d45007c6d2c46564/io.github.hydrusnetwork.hydrus.yml#L55)

**Comment by TingPing** [comment]
> for Celluloid bot do updates for shared-modules it won't.

That is a big problem then.

> If the lack of auto updating dependecies is not possible for shared modules, then how about having a regular flatpak package that is auto updated and then synced here?

I'm not sure about your wording but there aren't "flatpak packages" really.

**Comment by orowith2os** [comment]
If someone would like to take over my auto updater work for shared-modules, that would solve the out-of-date problem here, thus only leaving the main point, "mpv is complex".

**Comment by Erick555** [comment]
No, that would solve only half of the out-of-date problem. The second part is that f-e-d-c [doesn't update submodules](https://github.com/flathub-infra/flatpak-external-data-checker/issues/188) so everyone would need to bump it manually.

**Comment by orowith2os** [comment]
It might be possible to get the bot to check for a submodule named `shared-modules` and update that to the latest, then the other half is solved.

**Comment by Erick555** [comment]
Not yet  - monorepo contains dozens of modules while app may use 1-2 of them. Bumping submodule for each commit would results with useless updates for maintainers and useless updates for end users. Making bot aware of subpaths is complicated as noted in the linked thread.

**Comment by TingPing** [comment]
> Making bot aware of subpaths is complicated as noted in the linked thread.

While extra work I don't think is that crazy. In the manifest a module is either a string or an object. For strings they map to paths. With some effort you can see if those paths changed.

**Comment by bbhtt** [comment]
Use `io.mpv.Mpv` as `base:` in your manifest, convergence maintenance and update efforts to that by helping the maintainer. shared-modules is not a good place for mpv.
